Subject: lirc driver build fixups
From: kraxel@suse.de
Patch-mainline: never, depends on lirc-drivers patch

Make the damn drivers build inside the kernel tree,
minor fixes, sysfs class adaptions.

Index: linux-2.6.12/drivers/char/lirc/lirc_igorplugusb.c
===================================================================
--- linux-2.6.12.orig/drivers/char/lirc/lirc_igorplugusb.c	2005-07-28 11:55:58.000000000 +0200
+++ linux-2.6.12/drivers/char/lirc/lirc_igorplugusb.c	2005-07-28 12:23:13.000000000 +0200
@@ -59,9 +59,10 @@
 #include <linux/poll.h>
 #include <linux/smp_lock.h>
 #include <linux/time.h>
-#include "drivers/kcompat.h"
-#include "drivers/lirc.h"
-#include "drivers/lirc_dev/lirc_dev.h"
+
+#include "kcompat.h"
+#include "lirc.h"
+#include "lirc_dev.h"
 
 #if !defined(KERNEL_2_5)
 #        define USB_CTRL_GET_TIMEOUT    5
Index: linux-2.6.12/drivers/char/lirc/lirc_atiusb.c
===================================================================
--- linux-2.6.12.orig/drivers/char/lirc/lirc_atiusb.c	2005-07-28 11:55:58.000000000 +0200
+++ linux-2.6.12/drivers/char/lirc/lirc_atiusb.c	2005-07-28 12:13:32.000000000 +0200
@@ -55,9 +55,9 @@
 #include <linux/wait.h>
 #include <linux/list.h>
 
-#include "drivers/lirc.h"
-#include "drivers/kcompat.h"
-#include "drivers/lirc_dev/lirc_dev.h"
+#include "lirc.h"
+#include "kcompat.h"
+#include "lirc_dev.h"
 
 #define DRIVER_VERSION		"0.4"
 #define DRIVER_AUTHOR		"Paul Miller <pmiller9@users.sourceforge.net>"
Index: linux-2.6.12/drivers/char/lirc/lirc_mceusb.c
===================================================================
--- linux-2.6.12.orig/drivers/char/lirc/lirc_mceusb.c	2005-07-28 11:55:58.000000000 +0200
+++ linux-2.6.12/drivers/char/lirc/lirc_mceusb.c	2005-07-28 12:26:13.000000000 +0200
@@ -114,9 +114,9 @@
 	static int debug = 0;
 #endif
 
-#include "drivers/kcompat.h"
-#include "drivers/lirc.h"
-#include "drivers/lirc_dev/lirc_dev.h"
+#include "kcompat.h"
+#include "lirc.h"
+#include "lirc_dev.h"
 
 /* Use our own dbg macro */
 #define dprintk(fmt, args...)                             \
Index: linux-2.6.12/drivers/char/lirc/lirc_gpio.c
===================================================================
--- linux-2.6.12.orig/drivers/char/lirc/lirc_gpio.c	2005-07-28 11:55:58.000000000 +0200
+++ linux-2.6.12/drivers/char/lirc/lirc_gpio.c	2005-07-28 12:31:09.000000000 +0200
@@ -47,13 +47,11 @@
 
 #if LINUX_VERSION_CODE < KERNEL_VERSION(2,4,0)
 #include "../drivers/char/bttv.h"
-#include "../drivers/char/bttvp.h"
 #else
 #include "../drivers/media/video/bttv.h"
-#include "../drivers/media/video/bttvp.h"
 #endif
 
-#if BTTV_VERSION_CODE < KERNEL_VERSION(0,7,45)
+#if 0 /* BTTV_VERSION_CODE < KERNEL_VERSION(0,7,45) */
 #error "*******************************************************"
 #error " Sorry, this driver needs bttv version 0.7.45 or       "
 #error " higher. If you are using the bttv package, copy it to "
@@ -61,8 +59,8 @@
 #error "*******************************************************"
 #endif
 
-#include "drivers/kcompat.h"
-#include "drivers/lirc_dev/lirc_dev.h"
+#include "kcompat.h"
+#include "lirc_dev.h"
 
 /* insmod parameters */
 static int debug = 0;
Index: linux-2.6.12/drivers/char/lirc/lirc_serial.c
===================================================================
--- linux-2.6.12.orig/drivers/char/lirc/lirc_serial.c	2005-07-28 11:55:58.000000000 +0200
+++ linux-2.6.12/drivers/char/lirc/lirc_serial.c	2005-07-28 12:32:00.000000000 +0200
@@ -63,7 +63,7 @@
 
 #include <linux/config.h>
 
-#if defined(CONFIG_SERIAL) || defined(CONFIG_SERIAL_8250)
+#if 0 /* defined(CONFIG_SERIAL) || defined(CONFIG_SERIAL_8250) */
 #warning "******************************************"
 #warning " Your serial port driver is compiled into "
 #warning " the kernel. You will have to release the "
@@ -96,9 +96,9 @@
 #include <asm/irq.h>
 #include <asm/fcntl.h>
 
-#include "drivers/lirc.h"
-#include "drivers/kcompat.h"
-#include "drivers/lirc_dev/lirc_dev.h"
+#include "lirc.h"
+#include "kcompat.h"
+#include "lirc_dev.h"
 
 #if defined(LIRC_SERIAL_SOFTCARRIER) && !defined(LIRC_SERIAL_TRANSMITTER)
 #warning "Software carrier only affects transmitting"
Index: linux-2.6.12/drivers/char/lirc/lirc_parallel.c
===================================================================
--- linux-2.6.12.orig/drivers/char/lirc/lirc_parallel.c	2005-07-28 11:55:58.000000000 +0200
+++ linux-2.6.12/drivers/char/lirc/lirc_parallel.c	2005-07-28 12:31:26.000000000 +0200
@@ -69,9 +69,9 @@
 #include <linux/poll.h>
 #include <linux/parport.h>
 
-#include "drivers/lirc.h"
-#include "drivers/kcompat.h"
-#include "drivers/lirc_dev/lirc_dev.h"
+#include "lirc.h"
+#include "kcompat.h"
+#include "lirc_dev.h"
 
 #include "lirc_parallel.h"
 
Index: linux-2.6.12/drivers/char/lirc/lirc_sir.c
===================================================================
--- linux-2.6.12.orig/drivers/char/lirc/lirc_sir.c	2005-07-28 11:55:58.000000000 +0200
+++ linux-2.6.12/drivers/char/lirc/lirc_sir.c	2005-07-28 12:27:41.000000000 +0200
@@ -48,7 +48,7 @@
  
 #include <linux/config.h>
 
-#if !defined(LIRC_ON_SA1100) && !defined(CONFIG_SERIAL_MODULE)
+#if 0 /* !defined(LIRC_ON_SA1100) && !defined(CONFIG_SERIAL_MODULE) */
 #warning "******************************************"
 #warning " Your serial port driver is compiled into "
 #warning " the kernel. You will have to release the "
@@ -89,9 +89,9 @@
 
 #include <linux/timer.h>
 
-#include "drivers/lirc.h"
-#include "drivers/lirc_dev/lirc_dev.h"
-#include "drivers/kcompat.h"
+#include "lirc.h"
+#include "lirc_dev.h"
+#include "kcompat.h"
 
 /* SECTION: Definitions */
 
Index: linux-2.6.12/drivers/char/lirc/lirc_bt829.c
===================================================================
--- linux-2.6.12.orig/drivers/char/lirc/lirc_bt829.c	2005-07-28 11:55:58.000000000 +0200
+++ linux-2.6.12/drivers/char/lirc/lirc_bt829.c	2005-07-28 12:14:00.000000000 +0200
@@ -32,8 +32,8 @@
 #include <linux/pci.h>
 #include <linux/delay.h>
 
-#include "drivers/kcompat.h"
-#include "drivers/lirc_dev/lirc_dev.h"
+#include "kcompat.h"
+#include "lirc_dev.h"
 
 static int poll_main(void);
 static int atir_init_start(void);
Index: linux-2.6.12/drivers/char/lirc/lirc_it87.c
===================================================================
--- linux-2.6.12.orig/drivers/char/lirc/lirc_it87.c	2005-07-28 11:55:58.000000000 +0200
+++ linux-2.6.12/drivers/char/lirc/lirc_it87.c	2005-07-28 12:24:06.000000000 +0200
@@ -62,9 +62,9 @@
 
 #include <linux/timer.h>
 
-#include "drivers/lirc.h"
-#include "drivers/lirc_dev/lirc_dev.h"
-#include "drivers/kcompat.h"
+#include "lirc.h"
+#include "lirc_dev.h"
+#include "kcompat.h"
 
 #include "lirc_it87.h"
 
Index: linux-2.6.12/drivers/char/lirc/lirc_i2c.c
===================================================================
--- linux-2.6.12.orig/drivers/char/lirc/lirc_i2c.c	2005-07-28 11:55:58.000000000 +0200
+++ linux-2.6.12/drivers/char/lirc/lirc_i2c.c	2005-07-28 12:29:49.000000000 +0200
@@ -54,8 +54,8 @@
 
 #include <asm/semaphore.h>
 
-#include "drivers/kcompat.h"
-#include "drivers/lirc_dev/lirc_dev.h"
+#include "kcompat.h"
+#include "lirc_dev.h"
 
 struct IR {
 	struct lirc_plugin l;
Index: linux-2.6.12/drivers/char/lirc/kcompat.h
===================================================================
--- linux-2.6.12.orig/drivers/char/lirc/kcompat.h	2005-07-28 11:55:58.000000000 +0200
+++ linux-2.6.12/drivers/char/lirc/kcompat.h	2005-07-28 12:29:19.000000000 +0200
@@ -106,6 +106,8 @@ struct class_simple 
 #endif
 #define EXPORT_NO_SYMBOLS
 
+#define WE_DONT_USE_LOCAL_OPEN_CLOSE 0
+
 #else  /* Kernel < 2.5.0 */
 
 static inline int try_module_get(struct module *module)
@@ -127,10 +129,6 @@ static inline void module_put(struct mod
 #define MODULE_PARM_DESC(x,y)
 #endif
 
-#ifndef MODULE_ALIAS_CHARDEV_MAJOR
-#define MODULE_ALIAS_CHARDEV_MAJOR(x)
-#endif
-
 #ifndef MODULE_DEVICE_TABLE
 #define MODULE_DEVICE_TABLE(x,y)
 #endif
Index: linux-2.6.12/drivers/char/lirc/lirc_dev.c
===================================================================
--- linux-2.6.12.orig/drivers/char/lirc/lirc_dev.c	2005-07-28 11:55:58.000000000 +0200
+++ linux-2.6.12/drivers/char/lirc/lirc_dev.c	2005-07-28 12:24:06.000000000 +0200
@@ -58,8 +58,8 @@
 #include <linux/device.h>
 #endif
 
-#include "drivers/kcompat.h"
-#include "drivers/lirc.h"
+#include "kcompat.h"
+#include "lirc.h"
 #include "lirc_dev.h"
 
 static int debug = 0;
@@ -100,7 +100,7 @@ static struct irctl irctls[MAX_IRCTL_DEV
 static struct file_operations fops;
 
 /* Only used for sysfs but defined to void otherwise */
-static struct class_simple *lirc_class;
+static struct class *lirc_class;
 
 /*  helper function
  *  initializes the irctl structure
@@ -131,7 +131,7 @@ static void cleanup(struct irctl *ir)
 #ifdef LIRC_HAVE_DEVFS_26
 	devfs_remove(DEV_LIRC "/%u", ir->p.minor);
 #endif
-	class_simple_device_remove(MKDEV(IRCTL_DEV_MAJOR, ir->p.minor));
+	class_device_destroy(lirc_class, MKDEV(IRCTL_DEV_MAJOR, ir->p.minor));
 
 	if (ir->buf != ir->p.rbuf){
 		lirc_buffer_free(ir->buf);
@@ -375,8 +375,8 @@ int lirc_register_plugin(struct lirc_plu
 			S_IFCHR|S_IRUSR|S_IWUSR,
 			DEV_LIRC "/%u", ir->p.minor);
 #endif
-	(void) class_simple_device_add(lirc_class, MKDEV(IRCTL_DEV_MAJOR, ir->p.minor),
-				       NULL, "lirc%u", ir->p.minor);
+	class_device_create(lirc_class, MKDEV(IRCTL_DEV_MAJOR, ir->p.minor),
+			    NULL, "lirc%u", ir->p.minor);
 
 	if(p->sample_rate || p->get_queue) {
 		/* try to fire up polling thread */
@@ -408,7 +408,7 @@ int lirc_register_plugin(struct lirc_plu
 	return minor;
 	
 out_sysfs:
-	class_simple_device_remove(MKDEV(IRCTL_DEV_MAJOR, ir->p.minor));
+	class_device_destroy(lirc_class, MKDEV(IRCTL_DEV_MAJOR, ir->p.minor));
 #ifdef LIRC_HAVE_DEVFS_24
 	devfs_unregister(ir->devfs_handle);
 #endif
@@ -844,7 +844,7 @@ static int lirc_dev_init(void)
 		goto out;
 	}
 
-	lirc_class = class_simple_create(THIS_MODULE, "lirc");
+	lirc_class = class_create(THIS_MODULE, "lirc");
 	if(IS_ERR(lirc_class)) {
 		printk(KERN_ERR "lirc_dev: class_simple_create failed\n");
 		goto out_unregister;
@@ -884,7 +884,7 @@ void cleanup_module(void)
 	int ret;
 
 	ret = unregister_chrdev(IRCTL_DEV_MAJOR, IRCTL_DEV_NAME);
-	class_simple_destroy(lirc_class);
+	class_destroy(lirc_class);
 
 	if(ret)
 		printk("lirc_dev: error in module_unregister_chrdev: %d\n", ret);
Index: linux-2.6.12/drivers/char/lirc/lirc_sasem.c
===================================================================
--- linux-2.6.12.orig/drivers/char/lirc/lirc_sasem.c	2005-07-28 11:55:58.000000000 +0200
+++ linux-2.6.12/drivers/char/lirc/lirc_sasem.c	2005-07-28 13:10:15.000000000 +0200
@@ -61,8 +61,8 @@
 #include <linux/devfs_fs_kernel.h>
 
 #include "lirc_sasem.h"
-#include "drivers/lirc.h"
-#include "drivers/lirc_dev/lirc_dev.h"
+#include "lirc.h"
+#include "lirc_dev.h"
 
 //#define dbg(format, arg...) printk(KERN_DEBUG "%s: " format "\n" , __FILE__ , ## arg)
 
Index: linux-2.6.12/drivers/char/lirc/lirc_streamzap.c
===================================================================
--- linux-2.6.12.orig/drivers/char/lirc/lirc_streamzap.c	2005-07-28 11:55:58.000000000 +0200
+++ linux-2.6.12/drivers/char/lirc/lirc_streamzap.c	2005-07-28 13:09:29.000000000 +0200
@@ -49,9 +49,9 @@
 #include <asm/uaccess.h>
 #include <linux/usb.h>
 
-#include "drivers/lirc.h"
-#include "drivers/kcompat.h"
-#include "drivers/lirc_dev/lirc_dev.h"
+#include "lirc.h"
+#include "kcompat.h"
+#include "lirc_dev.h"
 
 #define DRIVER_VERSION	"$Revision: 1.10 $"
 #define DRIVER_NAME	"lirc_streamzap"
Index: linux-2.6.12/drivers/char/lirc/lirc_imon.c
===================================================================
--- linux-2.6.12.orig/drivers/char/lirc/lirc_imon.c	2005-07-28 11:55:58.000000000 +0200
+++ linux-2.6.12/drivers/char/lirc/lirc_imon.c	2005-07-28 13:09:47.000000000 +0200
@@ -60,9 +60,9 @@
 #include <linux/usb.h>
 #include <linux/devfs_fs_kernel.h>
 
-#include "drivers/lirc.h"
-#include "drivers/kcompat.h"
-#include "drivers/lirc_dev/lirc_dev.h"
+#include "lirc.h"
+#include "kcompat.h"
+#include "lirc_dev.h"
 
 
 #define MOD_AUTHOR	"Venky Raju <dev@venky.ws>"
Index: linux-2.6.12/drivers/char/lirc/lirc_sasem.h
===================================================================
--- linux-2.6.12.orig/drivers/char/lirc/lirc_sasem.h	2005-07-28 11:55:58.000000000 +0200
+++ linux-2.6.12/drivers/char/lirc/lirc_sasem.h	2005-07-28 13:10:00.000000000 +0200
@@ -3,7 +3,7 @@
 #ifndef LIRC_SASEM_H
 #define LIRC_SASEM_H
 
-#include "drivers/kcompat.h"
+#include "kcompat.h"
 
 /*
  * Version Information
